### PR TITLE
feature/DC299: Generate new Content type.

### PR DESCRIPTION
**Pull request by cosmicdreams**
- Add new generator and command to create new content types based on sample config.
- Solution for https://github.com/hechoendrupal/DrupalConsole/issues/299

### CODE REVIEW

File: src/Generator/GeneratorInterface.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Drupal\AppConsole\Generator;
+
+interface GeneratorInterface {
+
+  public function generate($module, $class_name, $routes, $test, $services, $class_machine_name);
+}
```
Comment (heddn) [inline src/Generator/GeneratorInterface.php]: Caught. Missing whitespace.

File: src/Generator/ContentTypeGenerator.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * Generate node.type.{ bundle_name }.yml
+     */
+    $this->renderFile(
+      'module/src/Entity/Bundle/node.type.yml.twig',
+      $this->getSite()->getModulePath($module) . '/config/install/node.type.' . $bundleName . '.yml',
+      $parameters
+    );
+  }
+
+}
```
Comment (heddn) [inline src/Generator/ContentTypeGenerator.php]: Caught again.

File: templates/module/src/Entity/Bundle/node.type.yml.twig [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    module:
+        - menu_ui
+third_party_settings:
+    menu_ui:
+        available_menus:
+            - main
+        parent: 'main:'
+name: {{ bundle_title }}
+type: {{ bundle_name }}
+description: 'Generated by Drupal Console'
```
Comment (heddn) [inline templates/module/src/Entity/Bundle/node.type.yml.twig]: Are quotes usually used in yml files? I seem to recall they weren't used. Please educate me.

Comment (seanr) [inline templates/module/src/Entity/Bundle/node.type.yml.twig]: They would indeed seem to be superfluous here, but not always - see http://stackoverflow.com/a/22235064/340462

**Pull request merged**